flipdisc/framework/app.py

```diff
@@ -502,25 +530,14 @@ def _on_redis_message(self, msg):
             self._log.debug("Redis sub callback")
             self._redis_sub_callback(app=self, channel=msg_channel, update=data)
 
-    def _on_redis_connect(self, sub_channels=None):
-        if "redis" in self.config:
-            db = self.config["redis"].get("db")
-            if db:
-                self._red.select(db)
-            pwd = self.config["redis"].get("auth")
-            if pwd:
-                self._red.auth(pwd)
-
-        if sub_channels:
-            self._red_sub.subscribe(sub_channels, self._on_redis_message)
-
     class ScheduledFunction:
-        def __init__(self, function, delay, periodic, loop):
+        def __init__(self, function, delay, periodic, loop, app):
             self.delay = delay
             self.function = function
             self.periodic = periodic
             self.started = False
             self._loop = loop
+            self._app = app
             self._task = None
```

Comment:
This is a minor thing, but would it make more sense to make this class take a generic `args` parameter instead of `app`, then use `self.function(*self.args)` down below when calling it?

```suggestion
    class ScheduledFunction:
        def __init__(self, function, args, delay, periodic, loop):
            self.delay = delay
            self.function = function
            self.args = args
            self.periodic = periodic
            self.started = False
            self._loop = loop
            self._task = None
```

Reply:
yeah for sure, might be nice to have the ability to pass other stuff